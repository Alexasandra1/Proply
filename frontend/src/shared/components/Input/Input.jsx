import React from "react"
import './Input.scss'

export function Input(props) {
    return (
        <div className="input">
            <input
                className={`input__input ${props.styleInput}`}
                placeholder={props.placeholder}
                value={props.value}
                onChange={props.onChange}
                type={props.type}
                pattern={props.pattern}
            />
        </div>
    )
}
