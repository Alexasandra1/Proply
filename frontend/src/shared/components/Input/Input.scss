.logInPage__body__form__password,
.logInPage__body__form__login, 
.registrationPage__body__form__name,
.registrationPage__body__form__login,
.registrationPage__body__form__password,
.registrationPage__body__form__city,
.registrationPage__body__form__email,
    .registrationPage__body__form__number,
    .addHousePage__body__form__designName,
    .addHousePage__body__form__price,
    .addHousePage__body__form__description,
    .addHousePage__body__form__car,
    .addHousePage__body__form__bed,
    .addHousePage__body__form__bath,
    .addHousePage__body__form__addres{
    height: 4.625rem;
    color: rgba(45, 124, 131, 0.5);
    // text-align: center;
    font-family: Montserrat;
    font-size: 2rem;
    border: 3px solid #2D7C83;
    margin-bottom: 2.06rem;
}


.forgotForm{
    width: 100%;
    height: 4.625rem;
    color: rgba(45, 124, 131, 0.5);
    // text-align: center;
    font-family: Montserrat;
    font-size: 2rem;
    border: 3px solid #2D7C83;
    margin-bottom: 2.06rem;
}

// .logInPage__body__form__login {
//     margin-bottom: 33px;
// }

// .logInPage__body__form__password {
//     margin-bottom: 51px;
// }


.input__input::placeholder {
    color: rgba(142, 186, 190, 0.5);
    font-weight: 1000;
    text-align: left;
  }



  @media (max-width: 420px) {
    .addHousePage__body__form__designName,
    .addHousePage__body__form__price,
    .addHousePage__body__form__description,
    .addHousePage__body__form__car,
    .addHousePage__body__form__bed,
    .addHousePage__body__form__bath,
    .addHousePage__body__form__addres{
        width: 90%;
    }
  }