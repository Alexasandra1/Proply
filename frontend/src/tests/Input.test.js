import React from 'react';
import renderer from 'react-test-renderer';
import { Input } from '../shared/components/Input/Input';

describe('Input', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Input styleInput="style" placeholder="Placeholder" value="Value" onChange={() => {}} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
