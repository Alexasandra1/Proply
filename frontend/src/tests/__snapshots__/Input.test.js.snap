// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div
  className="input"
>
  <input
    className="input__input style"
    onChange={[Function]}
    placeholder="Placeholder"
    value="Value"
  />
</div>
`;
